### PR TITLE
feat(modal-box): added support for title icon, modal alert states

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/3374

### CODE REVIEW

**Review by mmenestr — APPROVED** [review]
LGTM!

**Review by mcarrano — CHANGES_REQUESTED** [review]
Looks good @mcoker But maybe change the title of the last example to read "Error alert modal title" for consistency with the others?

**Comment by mcoker** [comment]
@mcarrano updated!

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks @mcoker !

**Review by mattnolting — APPROVED** [review]
The spacing between the icon and title text looks a bit tight to me, but implementation is sound. @mcarrano @mceledonia Is the spacing as expected?

File: src/patternfly/components/ModalBox/modal-box.scss [hunk 1]
```diff
@@ -84,6 +95,26 @@
...
+
+  &.pf-m-warning {
+    --pf-c-modal-box__title-icon--Color: var(--pf-c-modal-box--m-warning__title-icon--Color);
+  }
+
+  &.pf-m-success {
+    --pf-c-modal-box__title-icon--Color: var(--pf-c-modal-box--m-success__title-icon--Color);
+  }
+
+  &.pf-m-default {
```
Comment (christiemolloy) [inline src/patternfly/components/ModalBox/modal-box.scss:110]: do you need a modifier for default variation?

**Comment by mmenestr** [comment]
Yes, good catch @mattnolting. I initially had it mocked up as 16px, not 8px (as it appears to be but not sure).

File: src/patternfly/components/ModalBox/modal-box-title-icon.hbs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  {{#if modal-box-title-icon--attribute}}
+    {{{modal-box-title-icon--attribute}}}
+  {{/if}}>
+  {{#if modal-box--IsDefaultAlert}}
+    <i class="fas fa-bell" aria-hidden="true"></i>
+  {{else if modal-box--IsInfoAlert}}
+    <i class="fas fa-info-circle" aria-hidden="true"></i>
+  {{else if modal-box--IsSuccessAlert}}
+    <i class="fas fa-check-circle" aria-hidden="true"></i>
+  {{else if modal-box--IsWarningAlert}}
```
Comment (christiemolloy) [inline src/patternfly/components/ModalBox/modal-box-title-icon.hbs:11]: should this use fa-fw so all the icons line up?

**Review by christiemolloy — CHANGES_REQUESTED** [review]

File: src/patternfly/components/ModalBox/modal-box.scss (hunk 1 above)
Comment (mcoker) [reply src/patternfly/components/ModalBox/modal-box.scss:110]: I think so. Without a modifier, the icon is just a regular text color icon. With an alert modifier, it represents the alert state, and "default" is an alert state. WDYT?

File: src/patternfly/components/ModalBox/modal-box-title-icon.hbs (hunk 2 above)
Comment (mcoker) [reply src/patternfly/components/ModalBox/modal-box-title-icon.hbs:11]: updated!

**Review by mmenestr — APPROVED** [review]

**Review by christiemolloy — APPROVED** [review]
LPTM

**Pull request merged**